### PR TITLE
tolerate more than one gvklist item

### PR DESCRIPTION
**Pull request by juanvallejo**
Some third-party resources could be part of more than one api group.
Allow this to be the case when adding openapi models to openapi data,
and default to the first item as the gvk key for that model.

Related downstream issue: https://github.com/openshift/origin/issues/17872

**Release note**:
```release-note
NONE
```
cc @deads2k @soltysh

### CODE REVIEW

**Review by soltysh — APPROVED** [review]
/lgtm

**Comment by soltysh** [comment]
/unassign apelisse
/assign @deads2k 
for approval
/ok-to-test

**Comment by deads2k** [comment]
The rest of the code looks off too.  Why isn't the slice plumbed all the way up?

**Comment by juanvallejo** [comment]
@deads2k 
> The rest of the code looks off too. Why isn't the slice plumbed all the way up?

Not sure what you mean. Should we be parsing the gvk list in the caller of `parseGroupVersionKind`?

**Comment by juanvallejo** [comment]
@deads2k thanks for the feedback, switched to returning a slice of all found gvks

File: pkg/kubectl/cmd/util/openapi/openapi.go [hunk 1]
```diff
@@ -77,48 +79,53 @@ func (d *document) LookupResource(gvk schema.GroupVersionKind) proto.Schema {
...
+	// gvk extension must be a list of at least 1 element.
 	gvkList, ok := gvkExtension.([]interface{})
 	if !ok {
-		return schema.GroupVersionKind{}
+		return gvkListResult
 	}
-	if len(gvkList) != 1 {
-		return schema.GroupVersionKind{}
+	if len(gvkList) == 1 {
+		return gvkListResult
```
Comment (deads2k) [inline pkg/kubectl/cmd/util/openapi/openapi.go]: this looks wrong.  The entire block should go

File: pkg/kubectl/cmd/util/openapi/openapi.go [hunk 2]
```diff
@@ -77,48 +79,53 @@ func (d *document) LookupResource(gvk schema.GroupVersionKind) proto.Schema {
...
-		return schema.GroupVersionKind{}
+		return gvkListResult
 	}
 
-	// gvk extension must be a list of 1 element.
+	// gvk extension must be a list of at least 1 element.
 	gvkList, ok := gvkExtension.([]interface{})
 	if !ok {
-		return schema.GroupVersionKind{}
+		return gvkListResult
```
Comment (deads2k) [inline pkg/kubectl/cmd/util/openapi/openapi.go]: When you have a failure, you should generally return an empty or nil result.  empty will do.  Even though this *is* empty, the intent does not look empty.

File: pkg/kubectl/cmd/util/openapi/openapi.go [hunk 3]
```diff
@@ -77,48 +79,53 @@ func (d *document) LookupResource(gvk schema.GroupVersionKind) proto.Schema {
...
+func parseGroupVersionKind(s proto.Schema) []schema.GroupVersionKind {
 	extensions := s.GetExtensions()
 
+	gvkListResult := []schema.GroupVersionKind{}
+
 	// Get the extensions
 	gvkExtension, ok := extensions[groupVersionKindExtensionKey]
 	if !ok {
-		return schema.GroupVersionKind{}
+		return gvkListResult
```
Comment (deads2k) [inline pkg/kubectl/cmd/util/openapi/openapi.go]: explicit empty

File: pkg/kubectl/cmd/util/openapi/openapi.go (hunk 1 above)
Comment (juanvallejo) [reply pkg/kubectl/cmd/util/openapi/openapi.go]: Ah meant to change to `== 0`, but I agree, this block does not add anything now; will remove

File: pkg/kubectl/cmd/util/openapi/openapi.go (hunk 2 above)
Comment (juanvallejo) [reply pkg/kubectl/cmd/util/openapi/openapi.go]: Thanks, will change to actually return an empty slice of GroupVersionKind

**Comment by juanvallejo** [comment]
@deads2k thanks; comments addressed

**Pull request merged**